dataprovider/providers/cloud/data_provider.go

```diff
@@ -60,25 +72,43 @@ func (a DataProvider) FetchData(_ string, id string) (types.Data, error) {
 }
 
 func (a DataProvider) EnrichEvent(event *beat.Event, resMetadata fetching.ResourceMetadata) error {
-	_, err := event.Fields.Put(cloudAccountIdField, strings.FirstNonEmpty(resMetadata.AwsAccountId, a.accountId))
+	err := insertIfNotEmpty(cloudAccountIdField, strings.FirstNonEmpty(resMetadata.AwsAccountId, a.accountId), event)
```

Comment:
Blocker: make sure this doesn't break aws orgs account fields (LGTM but let's double check)

Reply:
@orestisfl how this can be validated?

Reply:
Right now, manually since we don't have E2E tests yet.

However, the unit tests cover the case already so maybe it's overkill.